internal/handlers/common/filter.go

```diff
@@ -1084,3 +1091,20 @@ func filterFieldValueByTypeCode(fieldValue any, code typeCode) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldExprElemMatch handles {field: {$elemMatch: value}}.
+// Returns false if doc value is not an array.
+func filterFieldExprElemMatch(doc *types.Document, filterKey string, exprValue any) (bool, error) {
```

Comment:
Should we have anything special about `$text` and `$where` not being supported in `$elemMatch`?

Reply:
Added correct response for `$text` and `$where`.